onsset/onsset.py

```diff
@@ -574,76 +596,61 @@ def grid_penalties(self):
         #define classifiers
         road_distance_labels = [5,4,3,2,1]
 
-        self.df[SET_ROAD_DIST_CLASSIFIED] = pd.cut(self.df[SET_ROAD_DIST], road_distance_bins,
-                                            labels=road_distance_labels).astype(float)
+        ROAD_DIST_CLASSIFIED = pd.cut(data_frame[SET_ROAD_DIST], road_distance_bins,
+                                            labels=road_distance_labels, include_lowest=True).astype(float)
 
         logging.info('Classify substation dist')
-        #define bins as -inf to 0.5, 0.5 to 1, 1 to 5, 5 to 10, 10 to inf
-        substation_distance_bins = [float("-inf"),0.5,1,5,10,float("inf")]
+        #define bins as 0 to 0.5, 0.5 to 1, 1 to 5, 5 to 10, 10 to inf
+        substation_distance_bins = [0,0.5,1,5,10,float("inf")]
         #define classifiers
         substation_distance_labels = [5,4,3,2,1]
 
-        self.df[SET_SUBSTATION_DIST_CLASSIFIED] = pd.cut(self.df[SET_SUBSTATION_DIST], substation_distance_bins,
+        SUBSTATION_DIST_CLASSIFIED = pd.cut(data_frame[SET_SUBSTATION_DIST], substation_distance_bins,
                                                   labels=substation_distance_labels).astype(float)
 
         logging.info('Classify land cover')
 
-        def classify_land_cover(data_frame,column,target_column):
-            """this is a different method employed to classify land cover and create new columns with the classification
-
-            Arguments
-            ---------
-            data_frame : input file
-            column : list
-            target_colum : list
-
-            """
-            data_frame.loc[(data_frame[column]==7) | (data_frame[column]==9 ) | (data_frame[column]==10 ) |
-                            (data_frame[column]==14 ) | (data_frame[column]==16 ) ,target_column] = 5
-            data_frame.loc[(data_frame[column]==2) | (data_frame[column]==4 ),target_column] = 4
-            data_frame.loc[(data_frame[column]==1 )| (data_frame[column]==3 ) | (data_frame[column]==5 ) |
-                            (data_frame[column]==12 ) | (data_frame[column]==13 ) | (data_frame[column]==15 ) ,target_column] = 3
-            data_frame.loc[(data_frame[column]==6) | (data_frame[column]==8 ) ,target_column] = 2
-            data_frame.loc[(data_frame[column]==0) | (data_frame[column]==11 ) ,target_column] = 1
-
-        classify_land_cover(self.df,SET_LAND_COVER,SET_LAND_COVER_CLASSIFIED)
+       
+        LAND_COVER_CLASSIFIED=self.classify_land_cover(data_frame[SET_LAND_COVER])
 
         logging.info('Classify elevation')
         #define bins as -inf to 500, 500 to 1000, 1000 to 2000, 2000 to 3000, 3000 to inf
         elevation_bins = [float("-inf"),500,1000,2000,3000,float("inf")]
         #define classifiers
         elevation_labels = [5,4,3,2,1]
 
-        self.df[SET_ELEVATION_CLASSIFIED] = pd.cut(self.df[SET_ELEVATION], elevation_bins,
+        ELEVATION_CLASSIFIED = pd.cut(data_frame[SET_ELEVATION], elevation_bins,
```

Comment:
Create function for elevation classification as I have shown for the road distance.